lib/tool_shed/util/shed_util_common.py

```diff
@@ -365,6 +365,21 @@ def get_tool_dependency_definition_metadata_from_tool_shed( app, tool_shed_url,
     return metadata
 
 
+def get_metadata_changeset_revisions( repository, repo ):
+    """
+    Return an unordered list of changeset_revisions and changeset numbers that are defined as installable.
+    """
+    changeset_tups = []
+    for repository_metadata in repository.downloadable_revisions:
+        ctx = hg_util.get_changectx_for_changeset( repo, repository_metadata.changeset_revision )
+        if ctx:
+            rev = ctx.rev()
+        else:
+            rev = -1
+        changeset_tups.append( ( rev, repository_metadata.changeset_revision ) )
+    return sorted( changeset_tups )
```

Comment:
`get_metadata_revisions()` method in this same file seems to already do the same (with more options).